dotnet/src/Connectors/Connectors.Memory.Qdrant/QdrantVectorStoreRecordMapper.cs

```diff
@@ -0,0 +1,299 @@
+        VectorStoreRecordPropertyReader.VerifyPropertyTypes(properties.vectorProperties, s_supportedVectorTypes, "Vector");
+
+        this._keyPropertyInfo = properties.keyProperty;
+        this._payloadPropertiesInfo = properties.dataProperties;
+        this._vectorPropertiesInfo = properties.vectorProperties;
+    }
+
+    /// <inheritdoc />
+    public PointStruct MapFromDataToStorageModel(TRecord dataModel)
+    {
+        PointId pointId;
+        if (this._keyPropertyInfo.PropertyType == typeof(ulong))
+        {
+            var key = this._keyPropertyInfo.GetValue(dataModel) as ulong? ?? throw new VectorStoreRecordMappingException($"Missing key property {this._keyPropertyInfo.Name} on provided record of type {typeof(TRecord).FullName}.");
+            pointId = new PointId { Num = key };
+        }
+        else if (this._keyPropertyInfo.PropertyType == typeof(Guid))
+        {
+            var key = this._keyPropertyInfo.GetValue(dataModel) as Guid? ?? throw new VectorStoreRecordMappingException($"Missing key property {this._keyPropertyInfo.Name} on provided record of type {typeof(TRecord).FullName}.");
+            pointId = new PointId { Uuid = key.ToString("D") };
+        }
+        else
+        {
+            throw new VectorStoreRecordMappingException($"Unsupported key type {this._keyPropertyInfo.PropertyType.FullName} for key property {this._keyPropertyInfo.Name} on provided record of type {typeof(TRecord).FullName}.");
+        }
+
+        // Create point.
+        var pointStruct = new PointStruct
+        {
+            Id = pointId,
+            Vectors = new Vectors(),
+            Payload = { },
+        };
+
+        // Add point payload.
+        foreach (var payloadPropertyInfo in this._payloadPropertiesInfo)
+        {
+            var propertyName = VectorStoreRecordPropertyReader.GetSerializedPropertyName(payloadPropertyInfo);
+            var propertyValue = payloadPropertyInfo.GetValue(dataModel);
+            pointStruct.Payload.Add(propertyName, ConvertToGrpcFieldValue(propertyValue));
+        }
+
+        // Add vectors.
+        if (this._options.HasNamedVectors)
+        {
+            var namedVectors = new NamedVectors();
+            foreach (var vectorPropertyInfo in this._vectorPropertiesInfo)
+            {
+                var propertyName = VectorStoreRecordPropertyReader.GetSerializedPropertyName(vectorPropertyInfo);
+                var propertyValue = vectorPropertyInfo.GetValue(dataModel);
+                if (propertyValue is not null)
+                {
+                    var castPropertyValue = (ReadOnlyMemory<float>)propertyValue;
+                    namedVectors.Vectors.Add(propertyName, castPropertyValue.ToArray());
+                }
+            }
+
+            pointStruct.Vectors.Vectors_ = namedVectors;
+        }
+        else
+        {
+            var vectorPropertyInfo = this._vectorPropertiesInfo.First();
```

Comment:
It looks like `this._vectorPropertiesInfo` can't be null, but it can be empty, in this case `First` will throw `System.InvalidOperationException: Sequence contains no elements`, which won't provide enough information to the user. Should we make it `FirstOrDefault`, and then if `null`, throw our exception with details?

Reply:
I actually do the verification for this already in the FindProperties, so when it gets to here it will always have at least one.  I'll add a comment to clarify, since otherwise it can be confusing.

This is the snippet in FindProperties where I'm doing the checking for this:

```cs
        // Check that we have one vector property if we don't have named vectors.
        if (!supportsMultipleVectors && !singleVectorPropertyFound)
        {
            throw new ArgumentException($"No vector property configured for type {type.FullName}.");
        }
```